cclib/parser/orcaparser.py

```diff
@@ -49,10 +49,120 @@ def before_parsing(self):
     def extract(self, inputfile, line):
         """Extract information from the file object inputfile."""
 
-        #  extract the version number first
-        if "Program Version" in line:
+        # Extract the version number
+        if "Program Version" == line.strip()[:15]:
             self.metadata["package_version"] = line.split()[2]
 
+        # ================================================================================
+        #                                         WARNINGS
+        #                        Please study these warnings very carefully!
+        # ================================================================================
+        #
+        # Warning: TCutStore was < 0. Adjusted to Thresh (uncritical)
+        #
+        # WARNING: your system is open-shell and RHF/RKS was chosen
+        #   ===> : WILL SWITCH to UHF/UKS
+        #
+        #
+        # INFO   : the flag for use of LIBINT has been found!
+        #
+        # ================================================================================
+        if "WARNINGS" == line.strip():
+            self.skip_lines(inputfile, ['text', '=', 'blank'])
+            if 'warnings' not in self.metadata:
+                self.metadata['warnings'] = []
+            if 'info' not in self.metadata:
+                self.metadata['info'] = []
+
+            line = next(inputfile)
+            while line[0] != '=':
+                if line.lower()[:7] == 'warning':
+                    self.metadata['warnings'].append('')
+                    while len(line) > 1:
+                        self.metadata['warnings'][-1] += line[9:]
+                        line = next(inputfile)
+                elif line.lower()[:4] == 'info':
+                    self.metadata['info'].append('')
+                    while len(line) > 1:
+                        self.metadata['info'][-1] += line[9:]
+                        line = next(inputfile)
+                line = next(inputfile)
+
+        # ================================================================================
+        #                                        INPUT FILE
+        # ================================================================================
+        # NAME = input.dat
+        # |  1> %pal nprocs 4 end
+        # |  2> ! B3LYP def2-svp
+        # |  3> ! Grid4
+        # |  4>
+        # |  5> *xyz 0 3
+        # |  6>     O   0   0   0
+        # |  7>     O   0   0   1.5
+        # |  8> *
+        # |  9>
+        # | 10>                          ****END OF INPUT****
+        # ================================================================================
+        if "INPUT FILE" == line.strip():
```

Comment:
Could you also add `input_file` that contains the entire input as one string?

Reply:
At first I thought you meant to store the name of the input file (which I implemented). However, upon rereading I presume you mean actually save all of the input as a string. I've corrected it to this in the latest commit.

Reply:
Actually, now that you mention it, both would be good. Perhaps call it `input_file_name`? Maybe the other should then be `input_file_contents`.